playbooks/v2_deploy.yml

```diff
@@ -0,0 +1,87 @@
+---
+
+- name: v2 deploy
+  hosts: ofn_servers
+  remote_user: "{{ user }}"
+  become: yes
+  become_user: "{{ unicorn_user }}"
+
+  roles:
+    # Include and enable only
+    - role: maintenance_mode
+      become: yes
+      become_user: root
+
+  handlers:
+    - include: ../roles/shared_handlers/handlers/main.yml
+
+  tasks:
+    - block:
+      - name: stop unicorn
+        service:
+          name: unicorn_{{ app }}
+          state: stopped
+        become: yes
+        become_user: root
+
+      - name: stop delayed job
+        service:
+          name: delayed_job_{{ app }}
+          state: stopped
+        become: yes
+        become_user: root
+
+      - include_role:
+          name: deploy
+
+      - meta: flush_handlers # Ensure handlers run successfully before reporting success
+      - name: refresh products cache
+        command: "{{ bundle_path }} exec rake ofn:cache:warm_products"
+        args:
+          chdir: "{{ current_path }}"
+
+      # Include and disable only
+      - name: disable maintenance mode
+        file:
+          dest: "/etc/nginx/maintenance.html"
```

Comment:
You shouldn't delete this file, it stays there permanently. To disable maintenance mode, run the role again with the `disable_maintenance` variable defined/true.